File: 예진희/3 JavaClass/210119_class_singleTone/singleTone/src/singleTone/SingleTest.java
package singleTone;

public class SingleTest {
	private static SingleTest inst; // 자기 자신의 타입
	private int data;

	public SingleTest() {
		data = (int) (Math.random() * 100);

	}
	
	
	//class method, 클래스 메소드
	public static SingleTest getInstance() { //리턴 타입이 SingleTest
		if (inst == null) {					 //처음에 null, 만들어진 객체를 ~ 계속 재사용 ~(계속 return), 
			inst = new SingleTest();
		}
		return inst;
	}
	
	public int getData() {
		return data;
	}
}

File: 장승민/CSharp/210215/singleTon/singleTon/SingleTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace singleTon
{
    class SingleTest
    {
        private static SingleTest inst;
        private int data;
        Random rd = new Random();
        public SingleTest()
        {
            data = (int)(rd.Next(0, 99));
        }
        public static SingleTest getInstance()
        {
            if (inst == null)
            {
                inst = new SingleTest();
            }
            return inst;
        }
        public int getData()
        {
            return data;
        }
    }
}
